src/sidebar/components/markdown-editor.js

```diff
@@ -134,69 +136,180 @@ ToolbarButton.propTypes = {
   label: propTypes.string,
   onClick: propTypes.func,
   shortcutKey: propTypes.string,
+  tabIndex: propTypes.string,
   title: propTypes.string,
 };
-
+/**
+ * An array of toolbar elements with a roving tab stop. Left and right
+ * array keys can be used to change focus of the elements. Home and end
+ * keys will navigate to the first and last elements respectively.
+ *
+ * Canonical example
+ * https://www.w3.org/TR/wai-aria-practices/examples/toolbar/toolbar.html
+ */
 function Toolbar({ isPreviewing, onCommand, onTogglePreview }) {
+  // Total number of elements in the toolbar minus 1.
+  const maxElementIndex = 9;
+
+  // Keep track of a roving index. The active roving tabIndex
+  // is set to 0, and all other elements are set to -1.
+  const [rovingElement, setRovingElement] = useState(0);
+
+  // An array of refs. Note we can't use `useRef` in a loop.
+  const elementsRefs = [];
+  for (let i = 0; i <= maxElementIndex; i++) {
+    elementsRefs.push(createRef());
```

Comment:
I could use some feedback on this. I think this is the "right" way to do it, but this is the first time I needed to have hooks, but in a loop. So instead of using useRef which won't work, I create an array of new refs.

Reply:
ya, good catch, clearly that was an issue, I suppose we could even use a useState here, but perhaps wrapping it in a ref is slightly more appropriate.